src/gui/reports/ReportsWidgetBrowserStatistics.cpp

```diff
@@ -320,23 +329,28 @@ void ReportsWidgetBrowserStatistics::saveSettings()
 
 void ReportsWidgetBrowserStatistics::deleteSelectedEntries()
 {
-    QList<Entry*> selectedEntries;
-    for (auto index : m_ui->browserStatisticsTableView->selectionModel()->selectedRows()) {
-        auto row = m_modelProxy->mapToSource(index).row();
-        auto entry = m_rowToEntry[row].second;
-        if (entry) {
-            selectedEntries << entry;
-        }
-    }
-
+    const auto& selectedEntries = getSelectedEntries();
     bool permanent = !m_db->metadata()->recycleBinEnabled();
+
     if (GuiTools::confirmDeleteEntries(this, selectedEntries, permanent)) {
         GuiTools::deleteEntriesResolveReferences(this, selectedEntries, permanent);
     }
 
     calculateBrowserStatistics();
 }
 
+void ReportsWidgetBrowserStatistics::deletePluginDataFromSelectedEntries()
+{
+    const auto& selectedEntries = getSelectedEntries();
+    if (GuiTools::confirmDeletePluginData(this, selectedEntries)) {
+        for (auto& entry : selectedEntries) {
+            entry->customData()->remove(BrowserService::KEEPASSXCBROWSER_NAME);
```

Comment:
This exposes too much browser plugin internals to the reports widget. Recommend switching to `browserService()->removePluginData(Entry*)` or similar